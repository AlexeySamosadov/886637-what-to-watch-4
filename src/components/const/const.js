export const AppRoute = {
  MAIN: `/`,
  FILM_INFO: `/films`,
  PLAYER: `player`,
  MY_LIST: `/my-list`,
  SIGN_IN: `/sign-in`,
  ADD_COMMENT: `review`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const ActiveMenu = {
  OVERVIEW: `OVERVIEW`,
  DETAILS: `DETAILS`,
  REVIEWS: `REVIEWS`,
};

export const RatingLevels = {
  AWESOME: `Awesome`,
  VERY_GOOD: `Very good`,
  GOOD: `Good`,
  NORMAL: `Normal`,
  BAD: `Bad`,
};

export const genreType = {
  ALL: `All genres`,
};

export const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

export const playerType = {
  MOVIE: `MOVIE`,
  TRAILER: `TRAILER`,
};

export const playerClass = {
  PLAYER_VIDEO: `player__video`,
};

export const keyCode = {
  ESCAPE: `Escape`,
  SPACE: `Space`,
  ARROW_RIGHT: `ArrowRight`,
  ARROW_LEFT: `ArrowLeft`,
};

export const typeEvent = {
  FULL_SCREEN_CHANGE: `fullscreenchange`,
  KEYDOWN: `keydown`,
};

export const arrowTimingPercent = {
  TEN: 3,
  MINUS_TEN: -3,
};

export const LENGTH = {
  MIN: 50,
  MAX: 400
};


