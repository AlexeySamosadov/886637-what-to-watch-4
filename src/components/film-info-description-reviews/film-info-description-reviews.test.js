import React from "react";
import renderer from "react-test-renderer";
import FilmInfoDescriptionReviews from "./film-info-description-reviews";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {ActiveMenu, GenreType} from "../const/const";

const moskStore = configureStore([]);

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const activeFilm = {
  backgroundColor: `#977461`,
  backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
  description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
  director: `Martin Scorsese`,
  genre: `Thriller`,
  id: 3,
  isFavorite: false,
  name: `Shutter Island`,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: 4.1,
  released: 2010,
  runTime: 138,
  scoresCount: 1002557,
  starring: [`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
};

const films = [activeFilm];

const store = moskStore({
  APP_STATUS: {
    activeMenuFilmInfo: ActiveMenu.OVERVIEW,
    activeGenre: GenreType.ALL,
    showingFilmsNumber: 8,
  },
  DATA: {
    films,
    promoFilm: activeFilm,
    reviews: [{
      comment: `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
      date: `2020-07-12T00:57:17.006Z`,
      id: 1,
      rating: 6.6,
      user: {id: 17, name: `Emely`}
    }, {
      comment: `I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.`,
      date: `2020-07-08T00:57:17.006Z`,
      id: 2,
      rating: 2.3,
      user: {id: 18, name: `Sophie`}
    }, {
      comment: `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
      date: `2020-07-28T00:57:17.006Z`,
      id: 3,
      rating: 7.7,
      user: {id: 14, name: `Corey`}
    }]
  },
  USER: {
    authorizationStatus: AuthorizationStatus.AUTH
  }
});

it(`Correctly render component FilmInfoDescriptionReviews`, () => {
  const tree = renderer.create(
      <Provider store={store}>
        <FilmInfoDescriptionReviews/>
      </Provider>, {
        createNodeMock: () => ({})
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
