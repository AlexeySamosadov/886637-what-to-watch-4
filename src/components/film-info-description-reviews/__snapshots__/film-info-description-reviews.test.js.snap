// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly render component FilmInfoDescriptionReviews 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Unfortunately we don't have a reliable way to tell the true ratings of a movie.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Emely
          </cite>
          <time
            className="review__date"
            dateTime="July 12, 2020"
          >
            July 12, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        6.6
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Sophie
          </cite>
          <time
            className="review__date"
            dateTime="July 8, 2020"
          >
            July 8, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        2.3
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Corey
          </cite>
          <time
            className="review__date"
            dateTime="July 28, 2020"
          >
            July 28, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7.7
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;
